src/main/io/gps.c

```diff
@@ -357,122 +425,128 @@ void gpsInit(void)
         return;
     }
 
-    while (gpsInitData[gpsData.baudrateIndex].baudrateIndex != gpsPortConfig->gps_baudrateIndex) {
-        gpsData.baudrateIndex++;
-        if (gpsData.baudrateIndex >= GPS_INIT_DATA_ENTRY_COUNT) {
-            gpsData.baudrateIndex = DEFAULT_BAUD_RATE_INDEX;
+    // set the user's intended baud rate
+    gpsData.userBaudRateIndex = 0;
+    initBaudRateIndex = BAUD_COUNT;
+    initBaudRateCycleCount = 0;
+    while (gpsInitData[gpsData.userBaudRateIndex].baudrateIndex != gpsPortConfig->gps_baudrateIndex) {
+        gpsData.userBaudRateIndex++;
+        if (gpsData.userBaudRateIndex >= GPS_INIT_DATA_ENTRY_COUNT) {
+            gpsData.userBaudRateIndex = DEFAULT_BAUD_RATE_INDEX; // 0, or fastest
             break;
         }
     }
+    // the user's intended baud rate will be used as the initial baud rate when connecting
+    gpsData.tempBaudRateIndex = gpsData.userBaudRateIndex;
```

Comment:
```suggestion
    initBaudRateIndex = BAUD_COUNT;
    initBaudRateCycleCount = 0;
    
    gpsData.userBaudRateIndex = DEFAULT_BAUD_RATE_INDEX;
    for (unsigned i = 0; i < GPS_INIT_DATA_ENTRY_COUNT; i++) {
      if (gpsInitData[i].baudrateIndex == gpsPortConfig->gps_baudrateIndex) {
        gpsData.userBaudRateIndex = i;
        break;
      }
    }
    // the user's intended baud rate will be used as the initial baud rate when connecting
    gpsData.tempBaudRateIndex = gpsData.userBaudRateIndex;
```

Reply:
Thank you, that is much nicer code.  Much appreciated.